python/data_structures/binary_search_tree.py

```diff
@@ -1,15 +1,52 @@
-from data_structures.binary_tree import BinaryTree
+from data_structures.binary_tree import BinaryTree, Node
 
 
 class BinarySearchTree(BinaryTree):
     """
     Put docstring here
     """
 
-    def __init__(self):
-        # initialization here
-        pass
+    def add(self, value):
+        def traverse(root, add_value):
+            if root == None:
+                return
+
+            if add_value.value < root.value:
+                if root.left != None:
+                    traverse(root.left, add_value)
+                else:
+                    root.left = add_value
+
+            else:
+                if root.right != None:
+                    traverse(root.right, add_value)
+                else:
+                    root.right = add_value
+
+        adding_node = Node(value)
+
+        if self.root == None:
+            self.root = adding_node
+        else:
+            traverse(self.root, adding_node)
+
+    def contains(self, search_value):
+        def traverse(root):
+            if not root:
+                return False
+
+            if search_value == root.value:
+                return True
+
+            if search_value > root.value:
+                traverse(root.right)
+
+            if search_value < root.value:
+                traverse(root.left)
+
+            return False
+
+        return traverse(self.root)
```

Comment:
Recursion can be tricky. Look closely at the `traverse` function here, more specifically where you `return`.

We have:
* return if the node doesn't exist -> False
* return if root value is what we're looking for -> True
* Final return after conditions evaluate -> False

The problem here is that on lines 42 and 45, you are calling your function recursively, and probably getting a return value from it, but not capturing or doing anything with that value as the DFS traverses down the tree. Right now your recursion happens, completes, and then the top-level call of `traverse` returns `False` no matter what via line 47, causing `contains` to then return `False` as well.

Hopefully that all made sense and can kind of point you in what direction this needs to go. :)